libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/MapStyleLoader.kt

```diff
@@ -0,0 +1,88 @@
+package com.mapbox.navigation.dropin
+
+import android.content.Context
+import android.content.res.Configuration
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import com.mapbox.maps.MapboxMap
+import com.mapbox.maps.Style
+import com.mapbox.maps.extension.observable.eventdata.StyleLoadedEventData
+import com.mapbox.maps.plugin.delegates.listeners.OnStyleLoadedListener
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.launch
+
+/**
+ * Lifecycle aware class that:
+ *  - loads initial map [Style] in onCreate
+ *  - reloads map [Style] on either [NavigationViewOptions.mapStyleUriNight] or
+ *   [NavigationViewOptions.mapStyleUriDay] flow changes
+ *  - registers itself as OnStyleLoadedListener and updates [loadedMapStyle] StateFlow value
+ *   every time new [Style] is fully loaded
+ */
+internal class MapStyleLoader(
+    private val context: Context,
+    private val mapboxMap: MapboxMap,
+    private val options: NavigationViewOptions,
+    private val loadedMapStyle: MutableStateFlow<Style?>,
```

Comment:
```suggestion
```

I don't think we should pass a MutableStateFlow in this way. The drop-in-ui has domain specific classes that wrap the lower level state flows.

What do you think about using the MapStyleLoader as a dependency for the UIComponents https://github.com/mapbox/mapbox-navigation-android/pull/5621/files#r837761035